### PR TITLE
Add ability to opt-out of access key localization

### LINKED ISSUE #2717: Add ability for locales to opt out of access key localization
Fluent Rich editor has a special UI for messages with access keys, which lists access key candidates.

Some locales don't translate access keys, mostly because they use non-Latin scripts.

We should add the ability for Locales to opt out of access key localization.

For those, we should use the original string as an access key candidate or even prefill the field with its value.

### PR DESCRIPTION
**Pull request by mathjazz**
Fix #2717.

This patch adds the ability for locales to opt-out of access key localization. For such locales, pretranslation keeps the accesskeys unchanged.

### CODE REVIEW

File: pontoon/pretranslation/pretranslate.py [hunk 1]
```diff
@@ -27,6 +27,16 @@ def __init__(self, locale):
         self.services = []
         self.locale = locale
 
+    def visit_Attribute(self, node):
+        is_accesskey = (
+            node.id.name.endswith("accesskey")
+            and isinstance(node.value.elements[0], ast.TextElement)
+            and len(node.value.elements[0].value) <= 1
```
Comment (mathjazz) [inline pontoon/pretranslation/pretranslate.py:34]: This part of the check is taken from the frontend:
https://github.com/mathjazz/pontoon/blob/2717-accesskey-opt-out/translate/src/modules/translationform/components/TranslationForm.tsx#L49

**Review by eemeli — CHANGES_REQUESTED** [review]
In principle this looks fine, but we need to figure out what's happening with #2837 until we land further changes in `pontoon/pretranslation/`.

File: pontoon/pretranslation/transformer.py [hunk 2]
```diff
@@ -135,6 +135,16 @@ def __init__(
         self.locale = locale
         self.services: list[str] = []
 
+    def visit_Attribute(self, node):
+        is_accesskey = (
+            node.id.name.endswith("accesskey")
+            and isinstance(node.value.elements[0], FTL.TextElement)
+            and len(node.value.elements[0].value) <= 1
```
Comment (eemeli) [inline pontoon/pretranslation/transformer.py:142]: This condition is either insufficient or too strict; we've plenty of accesskeys that are selectors on `PLATFORM()`. For example: https://pontoon.mozilla.org/en-GB/firefox/browser/browser/menubar.ftl/?string=220792

Given that this is gated on `accesskey_localization` which defaults to `True`, I don't think we need to look at the `.accesskey` value if the flag is set to `False`.

File: pontoon/pretranslation/transformer.py [hunk 3]
```diff
@@ -96,8 +96,8 @@ def create_locale_plural_variants(node: FTL.SelectExpression, locale: Locale):
 class PreparePretranslation(Transformer):
     """
     Flattens the given Pattern, uplifting selectors to the highest possible level and
-    duplicating shared parts in the variants. All other Placeables are serialised as
-    TextElements.
+    duplicating shared parts in the variants. Transforms plural variants to match the
+    locale. All other Placeables are serialised as TextElements.
```
Comment (eemeli) [inline pontoon/pretranslation/transformer.py:100]:
```suggestion
    locale.
```

Fixing my own mistake here. Nothing happens to the placeables in this transform.

**Review by eemeli — CHANGES_REQUESTED** [review]

**Review by eemeli — APPROVED** [review]

**Pull request merged**